src/main/java/com/xceptance/xlt/engine/htmlunit/okhttp3/OkHttp3WebConnection.java

```diff
@@ -206,7 +207,21 @@ protected Request createRequestWithStringBody(final URI uri, final WebRequest we
             contentType = (charset == null) ? mimeType : mimeType + ";charset=" + charset;
         }
 
-        final RequestBody requestBody = RequestBody.create(MediaType.get(contentType), body);
+        // create the request body
+        final RequestBody requestBody;
+
+        final MediaType mediaType = MediaType.get(contentType);
+        if (mediaType.charset() == null)
+        {
+            // interpret string body as binary/ISO_8859_1 content and create a binary request body
+            final byte[] bytes = body.getBytes(StandardCharsets.ISO_8859_1);
+            requestBody = RequestBody.create(mediaType, bytes);
+        }
```

Comment:
Shouldn't we check whether the user requested to send the request's data as binary at all? Otherwise, we might end up in sending UTF-8 encoded text as (incorrectly decoded) binary stream.

For example, imagine a PUT request w/ content-type ''text/json" and UTF-8 encoded JSON string as body.